lib/bacalhau.js

```diff
@@ -0,0 +1,162 @@
+import timers from 'node:timers/promises'
+import { execa } from 'execa'
+import { fetch } from 'undici'
+import * as Sentry from '@sentry/node'
+import { installBinaryModule, getBinaryModuleExecutable } from './modules.js'
+
+const DIST_TAG = 'v0.3.28'
+
+export async function install () {
+  await installBinaryModule({
+    module: 'bacalhau',
+    repo: 'bacalhau-project/bacalhau',
+    executable: 'bacalhau',
+    distTag: DIST_TAG,
+    targets: [
+      { platform: 'darwin', arch: 'x64', asset: `bacalhau_${DIST_TAG}_darwin_amd64.tar.gz` },
+      { platform: 'darwin', arch: 'arm64', asset: `bacalhau_${DIST_TAG}_darwin_arm64.tar.gz` },
+      { platform: 'linux', arch: 'x64', asset: `bacalhau_${DIST_TAG}_linux_amd64.tar.gz` },
+      { platform: 'linux', arch: 'arm64', asset: `bacalhau_${DIST_TAG}_linux_arm64.tar.gz` },
+      { platform: 'win32', arch: 'x64', asset: `bacalhau_${DIST_TAG}_windows_amd64.tar.gz` }
+    ]
+  })
+}
+
+export async function start ({
+  FIL_WALLET_ADDRESS,
+  storagePath,
+  metricsStream,
+  activityStream,
+  logStream
+}) {
+  logStream.write('Starting Bacalhau')
+
+  const childProcess = execa(
+    getBinaryModuleExecutable({
+      module: 'bacalhau',
+      executable: 'bacalhau'
+    }),
+    [
+      'serve',
+      '--log-mode=station',
+      '--node-type=compute',
+      '--limit-total-cpu=1',
+      '--limit-total-gpu=0',
+      '--limit-total-memory=200Mb'
+      // '--disable-engine=docker' (not yet released)
```

Comment:
I think we should not land this PR until this new flag is released. WDYT?

Reply:
It's still behind `--experimental`, I don't think this is a problem 🤔